lib/matplotlib/tests/test_figure.py

```diff
@@ -598,6 +600,14 @@ def test_invalid_layouts():
         fig.set_layout_engine("constrained")
 
 
+def test_layout_change_warning():
+    """Raise a warning when an existing layout manager is changed to tight."""
+    for layout in ('constrained', 'compressed'):
+        with pytest.warns(UserWarning):
+            fig, ax = plt.subplots(layout=layout)
```

Comment:
```suggestion
        fig, ax = plt.subplots(layout=layout)
        with pytest.warns(UserWarning, match="The layout manager has been"):
```
To check that it is the correct warning and clarify which operation warns.